runtime/runtime_linux.go

```diff
@@ -2,39 +2,220 @@ package runtime
 
 import (
 	"errors"
+	"fmt"
+	"io/ioutil"
 	"log"
+	"regexp"
+	"strings"
 	"time"
+
+	"github.com/taskcluster/generic-worker/host"
 )
 
 func (user *OSUser) CreateNew(okIfExists bool) (err error) {
-	return errors.New("(*(runtime.OSUser)).CreateNew(bool) not implemented on Linux")
+	if okIfExists {
+		panic("(*(runtime.OSUser)).CreateNew(true) not implemented on linux")
+	}
+
+	createUserScript := `
+		set -eu
+		username="${0}"
+		homedir="/home/${0}"
+		password="${1}"
+		echo "Creating user '${username}' with home directory '${homedir}' and password '${password}'..."
+		/usr/bin/sudo /usr/sbin/adduser --disabled-password --gecos "" --debug --home "${homedir}" "${username}"
+		echo "${username}:${password}" | /usr/bin/sudo /usr/sbin/chpasswd
+	`
+
+	return host.Run("/bin/bash", "-c", createUserScript, user.Name, user.Password)
 }
 
 func DeleteUser(username string) (err error) {
-	return errors.New("runtime.DeleteUser(string) not implemented on Linux")
+	return host.Run("/usr/bin/sudo", "/usr/sbin/deluser", "--force", "--remove-all-files", username)
 }
 
 func ListUserAccounts() (usernames []string, err error) {
-	return nil, errors.New("runtime.ListUserAccounts() not implemented on Linux")
+	var passwd []byte
+	passwd, err = ioutil.ReadFile("/etc/passwd")
+	if err != nil {
+		return
+	}
+	lines := strings.Split(string(passwd), "\n")
+	usernames = []string{}
+	for _, line := range lines {
+		if line != "" {
+			usernames = append(usernames, strings.SplitN(line, ":", 2)[0])
+		}
+	}
+	return
 }
 
 func UserHomeDirectoriesParent() string {
 	return "/home"
 }
 
 func WaitForLoginCompletion(timeout time.Duration) error {
```

Comment:
This timeout/sleep deadline works, though it might be more idiomatic to use a context here.

Reply:
Fair point, I'll see if I can make that more idiomatic. ++

Reply:
On reflection, `WaitForLoginCompletion` spans all platforms, so changing the method signature here would mean changing all platform implementations, which may be a bit of unnecessary scope creep for this PR, so I think I'll leave it as if for now, and another PR can overhaul the `WaitForLoginCompletion` method across all platforms (windows/linux/darwin) to be more idiomatic, if required.